### PR TITLE
Added a blog explaining how to send requests using JWT's to OpenID Connect

### PR DESCRIPTION
**Pull request by PrarthonaPaul**
https://prarthonapaul.github.io/wildfly-elytron/blog/oidc-cient-request/

### CODE REVIEW

File: _posts/2024-07-03-oidc-cient-request.adoc [hunk 1]
```diff
@@ -0,0 +1,440 @@
...
+
+. Navigate to that tab and click on *Import*. 
+* Under *Archive format*, choose *PKCS12* from the dropdown. 
+* Under *Key alias*,
+enter the alias of the RSA key you just created. 
+* Under *Store password* enter the password of the keystore you created
+and click on the *Browse* button to import the keystore file. Note that when we created the keystore, we specified a key
+password and a keystore password. Here, we are only using the *keystore* password. While we set them both to be the same, they do not have to be.
+
+. Finally, create a user called *alice* whose *First name* is *Alice* and *Last name* is *Smith*.
```
Comment (rsearls) [inline _posts/2024-07-03-oidc-cient-request.adoc]: Directions as to how to create user, alice in Keycloak is needed.

File: _posts/2024-07-03-oidc-cient-request.adoc [hunk 2]
```diff
@@ -0,0 +1,440 @@
...
+* Under *Archive format*, choose *PKCS12* from the dropdown. 
+* Under *Key alias*,
+enter the alias of the RSA key you just created. 
+* Under *Store password* enter the password of the keystore you created
+and click on the *Browse* button to import the keystore file. Note that when we created the keystore, we specified a key
+password and a keystore password. Here, we are only using the *keystore* password. While we set them both to be the same, they do not have to be.
+
+. Finally, create a user called *alice* whose *First name* is *Alice* and *Last name* is *Smith*. 
+
+. Next, navigate to the *Credentials* tab and create a password for Alice. Toggle the `Temporary` switch off, so you are
```
Comment (rsearls) [inline _posts/2024-07-03-oidc-cient-request.adoc]: I am stuck at this point and am unable to proceed.  I can not find the "Credentials" tab

File: _posts/2024-07-03-oidc-cient-request.adoc [hunk 3]
```diff
@@ -0,0 +1,440 @@
...
+```
+    $ keytool -genkeypair -alias <keystore_alias> -keyalg <algorithm> -keysize <key_size> -validity <validity_in_days>
+-keystore <keystore_name> -dname "<distinguished_name>" -keypass <private_key_password> -storepass <keystore_password>
+```
+
+==== Generating PKCS12 Keystore
+
+Let us first create a PKCS12 keystore file with 2 different keys. Let's first generate an RSA keyPair: 
+```
+    $ keytool -genkeypair -alias rsaKey -keyalg RSA -keysize 2048 -validity 365 -keystore Keycloak.keystore.pkcs12 -dname "CN=client" -keypass password -storepass password
```
Comment (rsearls) [inline _posts/2024-07-03-oidc-cient-request.adoc]: You should tell the user to make a note of the "alias", "storepass" password and the absolute path to the "keystore" file, Keycloak.keystore.pkcs12,  because this information must be provide to Keycloak in a future step.

File: _posts/2024-07-03-oidc-cient-request.adoc [hunk 4]
```diff
@@ -0,0 +1,440 @@
...
+```
+    $ keytool -list -v -keystore Keycloak.keystore.pkcs12 -storepass password
+```
+
+If you would like to use JKS keystore files instead, see https://docs.redhat.com/en/documentation/red_hat_jboss_data_virtualization/6.4/html/security_guide/create_a_privatepublic_key_pair_with_keytool[here] for more information.
+
+== Log Into the OpenShift Cluster
+Before we can deploy our application, we need to log into an OpenShift cluster. You can log in via the OpenShift CLI:
+```
+oc login -u myUserName
```
Comment (rsearls) [inline _posts/2024-07-03-oidc-cient-request.adoc]: Upon login a list of available projects are listed.   The user must designate the project to make active with cmd,
   oc project <PROJECT_NAME>

File: _posts/2024-07-03-oidc-cient-request.adoc [hunk 5]
```diff
@@ -0,0 +1,440 @@
...
+
+Once you have your environment set up with the required tools, we can move on to the next step to build and deploy our application on OpenShift.
+
+== Start Keycloak
+We will be using Keycloak as our OpenID identity provider.
+
+To start a Keycloak server in your project on OpenShift, use the following command:
+
+```
+oc process -f https://raw.githubusercontent.com/keycloak/keycloak-quickstarts/latest/openshift/keycloak.yaml \
```
Comment (rsearls) [inline _posts/2024-07-03-oidc-cient-request.adoc]: This cmd format did not work for me.   I had to remove the '\' and provide all text on a single line.
(oc process -f https://raw.githubusercontent.com/keycloak/keycloak-quickstarts/latest/openshift/keycloak.yaml  -p KEYCLOAK_ADMIN=admin  -p KEYCLOAK_ADMIN_PASSWORD=admin  -p NAMESPACE=<PROJECT_NAME>  oc create -f -  )

File: _posts/2024-07-03-oidc-cient-request.adoc [hunk 6]
```diff
@@ -0,0 +1,440 @@
...
+Once the Keycloak server has been provisioned, use the following command to find the URL for your Keycloak instance’s Admin Console:
+```
+KEYCLOAK_URL=https://$(oc get route keycloak --template='{{ .spec.host }}') &&
+echo "" &&
+echo "Keycloak Admin Console:   $KEYCLOAK_URL/admin" &&
+echo ""
+```
+
+=== OpenShift Web Console
+To make sure your Keycloak server has been provisioned using the OpenShift web console, navigate to the *Topology* view in the *Developer* perspective. You can click on your *keycloak* app to check its status. Once it is running, you can click on *Open URL* and then access Keycloak’s *Administration Console*.
```
Comment (rsearls) [inline _posts/2024-07-03-oidc-cient-request.adoc]: I recommend replacing, "and then access Keycloak’s *Administration Console*." with, " and you will be taken to  Keycloak’s *Administration Console*  login page"

File: _posts/2024-07-03-oidc-cient-request.adoc [hunk 7]
```diff
@@ -0,0 +1,440 @@
...
+** *Client type* (or *Client Protocol*, depending on your keycloak version): *OpenID Connect*
+** *Client ID*: *myclient*
+* *Capability config*:
+** *Client authentication* : *On*
+** *Authentication flow*: *Standard flow, Direct access grants*
+* *Login Settings*: For the *Valid Redirect URIs*, enter a * for now. We will come back to edit it later.
+
+. Once you hit *Save*, you should see a new tab called *Keys* appear.
+
+. Navigate to that tab and click on *Import*.
```
Comment (rsearls) [inline _posts/2024-07-03-oidc-cient-request.adoc]: say,  click on the *Import* button at the bottom of the page.

File: _posts/2024-07-03-oidc-cient-request.adoc [hunk 8]
```diff
@@ -0,0 +1,440 @@
...
+. Once you hit *Save*, you should see a new tab called *Keys* appear.
+
+. Navigate to that tab and click on *Import*. 
+* Under *Archive format*, choose *PKCS12* from the dropdown. 
+* Under *Key alias*,
+enter the alias of the RSA key you just created. 
+* Under *Store password* enter the password of the keystore you created
+and click on the *Browse* button to import the keystore file. Note that when we created the keystore, we specified a key
+password and a keystore password. Here, we are only using the *keystore* password. While we set them both to be the same, they do not have to be.
+
```
Comment (rsearls) [inline _posts/2024-07-03-oidc-cient-request.adoc]: It's unclear what to do after uploading the keystore via the brower.   The user is returned to the keys tab page.
The "save" button is disabled.  The "Generate new key" and "Import" buttons are active but neither make sense to select.

File: _posts/2024-07-03-oidc-cient-request.adoc (hunk 1 above)
Comment (PrarthonaPaul) [reply _posts/2024-07-03-oidc-cient-request.adoc]: Fixed!

File: _posts/2024-07-03-oidc-cient-request.adoc (hunk 2 above)
Comment (PrarthonaPaul) [reply _posts/2024-07-03-oidc-cient-request.adoc]: I have added more clear instructions for this.

File: _posts/2024-07-03-oidc-cient-request.adoc (hunk 3 above)
Comment (PrarthonaPaul) [reply _posts/2024-07-03-oidc-cient-request.adoc]: I added a note for this.

File: _posts/2024-07-03-oidc-cient-request.adoc (hunk 4 above)
Comment (PrarthonaPaul) [reply _posts/2024-07-03-oidc-cient-request.adoc]: Added a note for this

File: _posts/2024-07-03-oidc-cient-request.adoc (hunk 5 above)
Comment (PrarthonaPaul) [reply _posts/2024-07-03-oidc-cient-request.adoc]: right, I believe this is formatted like that to add the annotations <1>, <2> and so on. 
Some of the other guides also had the same formatting for this command. 
I have added `//` in front of the annotations to make that more clear.

File: _posts/2024-07-03-oidc-cient-request.adoc (hunk 6 above)
Comment (PrarthonaPaul) [reply _posts/2024-07-03-oidc-cient-request.adoc]: fixed!

File: _posts/2024-07-03-oidc-cient-request.adoc (hunk 7 above)
Comment (PrarthonaPaul) [reply _posts/2024-07-03-oidc-cient-request.adoc]: fixed!

File: _posts/2024-07-03-oidc-cient-request.adoc (hunk 8 above)
Comment (PrarthonaPaul) [reply _posts/2024-07-03-oidc-cient-request.adoc]: Added info about what to look for to ensure that the cert has been added correctly. 
Thanks @rsearls, I have addressed all the comments.

File: _posts/2024-07-03-oidc-cient-request.adoc [hunk 9]
```diff
@@ -0,0 +1,447 @@
...
+However, OpenID Connect allows the Request Object to be sent as a JWT using https://openid.net/specs/openid-connect-core-1_0.html#JWTRequests[request parameters], which can be signed and optionally
+encrypted. This adds an added layer of protection, as the parameters are no longer human-readable.
+
+toc::[]
+
+== Prerequisites 
+
+To follow along with this guide, you will need:
+
+* Roughly 15 minutes
```
Comment (rsearls) [inline _posts/2024-07-03-oidc-cient-request.adoc]: The time needed to setup the environment and run the test is considerably
longer than 15 minutes.  Setting up the environment for scratch takes an 
hour or more.  If you want to keep this statement I suggest stating, 
"Once your environment is setup running the test will take roughly 15 minutes".

File: _posts/2024-07-03-oidc-cient-request.adoc [hunk 10]
```diff
@@ -0,0 +1,447 @@
...
+
+* Access to an OpenShift cluster (try the Red Hat Developer Sandbox for free)
+
+* https://docs.openshift.com/container-platform/4.15/cli_reference/openshift_cli/getting-started-cli.html[OpenShift CLI]
+
+* https://helm.sh/docs/intro/install/[Helm Chart]
+
```
Comment (rsearls) [inline _posts/2024-07-03-oidc-cient-request.adoc]: I suggest adding a bullet item that provides the link to the test app to this list.

File: _posts/2024-07-03-oidc-cient-request.adoc [hunk 11]
```diff
@@ -0,0 +1,447 @@
...
+```
+oc login --token=myToken --server=myServerUrl
+```
+
+Once you log in, you will see a list of projects you have access to. You can switch between them to select the one where you would like to deploy your application using: 
+```
+oc project <projectname>
+```
+
+Once you have your environment set up with the required tools, we can move on to the next step to build and deploy our application on OpenShift.
```
Comment (rsearls) [inline _posts/2024-07-03-oidc-cient-request.adoc]: This statement should be moved to after the keycloak setup.

File: _posts/2024-07-03-oidc-cient-request.adoc [hunk 12]
```diff
@@ -0,0 +1,447 @@
...
+```
+
+Once the Keycloak server has been provisioned, use the following command to find the URL for your Keycloak instance’s Admin Console:
+```
+KEYCLOAK_URL=https://$(oc get route keycloak --template='{{ .spec.host }}') &&
+echo "" &&
+echo "Keycloak Admin Console:   $KEYCLOAK_URL/admin" &&
+echo ""
+```
+
```
Comment (rsearls) [inline _posts/2024-07-03-oidc-cient-request.adoc]: Tell the user to make a note of the "KEYCLOAK_URL".  The URL will be needed in the next step.

File: _posts/2024-07-03-oidc-cient-request.adoc [hunk 13]
```diff
@@ -0,0 +1,447 @@
...
+=== OpenShift Web Console
+To make sure your Keycloak server has been provisioned using the OpenShift web console, navigate to the *Topology* view in the *Developer* perspective. You can click on your *keycloak* app to check its status. Once it is running, you can click on *Open URL* and you will be taken to Keycloak’s *Administration Console* login page.
+
+== Setting up Keycloak OpenID Provider
+
+. Log into the Keycloak Admin Console using the admin username and password you specified earlier. 
+. Next, create a realm called *myrealm* and register a client called *myclient* as follows: 
+*General settings*:
```
Comment (rsearls) [inline _posts/2024-07-03-oidc-cient-request.adoc]: The current Keycloak version is 25.0.1.  The screen presentation appears differently from what you describe. 
There is no "General settings" displayed anywhere.   Here are directions for the current Keycloak UI.

-- Create a realm for our test
* Select the menu in the upper left corner
* Click on the drop down box with text "Keycloak"
* Click "Create realm" button
* In the "Realm name" field type, "myrealm"
* Click the "Create" button

-- Create client, myclient
* Select the menu in the upper left corner
* Select the "Client" menu item.
* On the Client page, select the "Create client" button
* Set the "Client type" field to *OpenID Connect*
* Set the "Client ID" to *myclient*
* Select the "Next" button
* On the "Capability config" screen
** Toggle the "Client authentication" to ON
** For "Authentication flow"  select checkboxes, *Standard flow*, and *Direct access grants*
** Click the "Next" button.
* On the "Login settings" screen
** For the *Valid Redirect URIs*, enter a * for now. We will come back to edit it later.
** Click the "Save" button
* On the "Client details" page for "myclient", click the "Keys" tab.
** click on the *Import* button located at the bottom of the screen.
*** Under *Archive format*, choose *PKCS12* from the dropdown. 
*** Under *Key alias*, enter the alias of the RSA key you just created. 
*** Under *Store password* enter the password of the keystore you created
    and click on the *Browse* button to import the keystore file. Note that 
    when we created the keystore, we specified a key
    password and a keystore password. Here, we are only using the 
    *keystore* password. While we set them both to be the same, they 
    do not have to be.
*** Once you select the keystore file created earlier, click *Import* 
    and you should see a message at the top of the screen indicating 
    that the certificate has been uploaded successfully and you will 
    see it in the text field in the middle of the screen.

File: _posts/2024-07-03-oidc-cient-request.adoc [hunk 14]
```diff
@@ -0,0 +1,447 @@
...
+
+. Navigate to that tab and click on the *Import* button located at the bottom of the screen. 
+* Under *Archive format*, choose *PKCS12* from the dropdown. 
+* Under *Key alias*, enter the alias of the RSA key you just created. 
+* Under *Store password* enter the password of the keystore you created
+and click on the *Browse* button to import the keystore file. Note that when we created the keystore, we specified a key
+password and a keystore password. Here, we are only using the *keystore* password. While we set them both to be the same, they do not have to be.
+* Once you select the keystore file created earlier, click *Import* and you should see a message at the top of the screen indicating that the certificate has been uploaded successfully and you will see it in the text field in the middle of the screen.
+
+. Finally, create a user called *alice* whose *First name* is *Alice* and *Last name* is *Smith*. You can find more details about creating and managing KeyCloak users https://www.keycloak.org/docs/latest/server_admin/#proc-creating-user_server_administration_guide[here].
```
Comment (rsearls) [inline _posts/2024-07-03-oidc-cient-request.adoc]: These directions will help the user create a new user more quickly.
-- create a new user for realm, myrealm
Select the drop down menu in the upper left corner
Select the "User" menu item.
On the "Users" screen click the "Create new user" button
In the "Username" box type the user alias, "alice"
In the "First name" box type "Alice"
In the "Last name" box type "Smith"
Click the "Create" button

-- set Alice's account password
On alice's "User details" page, click on the "Credentials" tab.
Click the "Set password" button.
In the "Set password" dialog, set a password of your choosing for alice.
Make note of this password.  It will be required later
Toggle the `Temporary` switch to off, so you are not prompted to update 
the password after the first login.
Click the "Save" button

File: _posts/2024-07-03-oidc-cient-request.adoc [hunk 15]
```diff
@@ -0,0 +1,447 @@
...
+    - name: request-object-signing-keystore-volume
+      secret:
+        secretName: simple-webapp-secret
+  volumeMounts:
+    - name: request-object-signing-keystore-volume
+      mountPath: /etc/request-object-secret-volume
+      readOnly: true
+```
+<1> Replace <KEYCLOAK_URL> with the Keycloak URL obtained in the previous command.
+<2> Replace <CLIENT_SECRET> with the client secret for `myclient`.
```
Comment (rsearls) [inline _posts/2024-07-03-oidc-cient-request.adoc]: I don't understand what to designate for the "CLIENT_SECRET".   Here was no previous direction to create a "client secrete".  There was a direction to create an "OpenShift secret".  There was not direction to save the secret text for future reference.

This is blocking me from proceeding with the example.

Comment (PrarthonaPaul) [reply _posts/2024-07-03-oidc-cient-request.adoc]: Thanks @rsearls 
I have addressed the comments till here.

File: _posts/2024-07-03-oidc-cient-request.adoc [hunk 16]
```diff
@@ -0,0 +1,467 @@
...
+However, OpenID Connect allows the Request Object to be sent as a JWT using https://openid.net/specs/openid-connect-core-1_0.html#JWTRequests[request parameters], which can be signed and optionally
+encrypted. This adds an added layer of protection, as the parameters are no longer human-readable.
+
+toc::[]
+
+== Prerequisites 
+
+To follow along with this guide, you will need:
+
+* Roughly 20 minutes (not including the time needed for building)
```
Comment (fjuma) [inline _posts/2024-07-03-oidc-cient-request.adoc]: Time for building what?

File: _posts/2024-07-03-oidc-cient-request.adoc [hunk 17]
```diff
@@ -0,0 +1,467 @@
...
+
+* https://helm.sh/docs/intro/install/[Helm Chart]
+
+* https://github.com/wildfly-security-incubator/elytron-examples[Elytron-examples project]
+
+== Example Application
+
+We will use a simple web application in this guide that consists of a https://github.com/wildfly-security-incubator/elytron-examples/blob/main/elytron-oidc-request/src/main/java/org/wildfly/security/examples/SecuredServlet.java[single servlet]. We will secure this servlet using OIDC.
+
+We will use the example in the https://github.com/wildfly-security-incubator/elytron-examples/tree/main/elytron-oidc-request[elytron-oidc-request] directory in this repo.
```
Comment (fjuma) [inline _posts/2024-07-03-oidc-cient-request.adoc]: s/in this repo/in the `elytron-examples` repo

File: _posts/2024-07-03-oidc-cient-request.adoc [hunk 18]
```diff
@@ -0,0 +1,467 @@
...
+
+To obtain this example, clone the elytron-examples repository to your local machine:
```
Comment (fjuma) [inline _posts/2024-07-03-oidc-cient-request.adoc]: `elytron-examples`

File: _posts/2024-07-03-oidc-cient-request.adoc [hunk 19]
```diff
@@ -0,0 +1,467 @@
...
+
+==== Generating PKCS12 Keystore
+
+Let us first create a PKCS12 keystore file with 2 different keys. Let's first generate an RSA keyPair: 
+```
+    $ keytool -genkeypair -alias rsaKey -keyalg RSA -keysize 2048 -validity 365 -keystore Keycloak.keystore.pkcs12 -dname "CN=client" -keypass password -storepass password
+```
+And for Elliptical Curve keys, use the following commands: 
+```
+    $ keytool -genkeypair -alias ecKey -keyalg EC -groupname secp256r1 -validity 365 -keystore Keycloak.keystore.pkcs12 -dname "CN=client" -keypass password -storepass password
```
Comment (fjuma) [inline _posts/2024-07-03-oidc-cient-request.adoc]: Is this key pair used in this example or is the goal to just show how to generate a key pair using EC?

File: _posts/2024-07-03-oidc-cient-request.adoc [hunk 20]
```diff
@@ -0,0 +1,467 @@
...
+
+* *Private Key:* This is the part of the key that must be kept secret and will be used to sign the JWT. Your OpenID provider communicates the public key of some keyPair(s) to you. The public key is used by you to encrypt the JWT.
+
+* *Public Key:* This is the part of the keypair that is shared with trusted parties. Typically, the OpenID provider shared a set of public keys with us which we use to encrypt our JWT's among other things. Public keys can be extracted from certificates.
+
+* *Certificate:* This is used to verify the signatures among other things. We share the certificates of keypairs we generate with the OpenID provider, and it contains the public key.
+
+Keypairs with public and private keys are known as *asymmetric* keys. Some keys are *symmetric*, meaning they only have 1 secret key that is used by both the client and the server. Each key has an algorithm that is used to generate it, such as, RSA, Elliptic Curve, etc. An example of a symmetric algorithm is HS256, where the same key is used to sign a JWT and verify the signature.
+
+=== Generating Key Pairs
```
Comment (fjuma) [inline _posts/2024-07-03-oidc-cient-request.adoc]: Are multiple key pairs needed for the example or just one?

File: _posts/2024-07-03-oidc-cient-request.adoc [hunk 21]
```diff
@@ -0,0 +1,467 @@
...
+
+=== About Key Pairs
+
+Typically key pairs have 3 components: 
+
+* *Private Key:* This is the part of the key that must be kept secret and will be used to sign the JWT. Your OpenID provider communicates the public key of some keyPair(s) to you. The public key is used by you to encrypt the JWT.
+
+* *Public Key:* This is the part of the keypair that is shared with trusted parties. Typically, the OpenID provider shared a set of public keys with us which we use to encrypt our JWT's among other things. Public keys can be extracted from certificates.
+
+* *Certificate:* This is used to verify the signatures among other things. We share the certificates of keypairs we generate with the OpenID provider, and it contains the public key.
```
Comment (fjuma) [inline _posts/2024-07-03-oidc-cient-request.adoc]: s/keypairs/key pairs (please update all references in this doc)

File: _posts/2024-07-03-oidc-cient-request.adoc [hunk 22]
```diff
@@ -0,0 +1,467 @@
...
+    "ssl-required" : "EXTERNAL",
+    "scope" : "profile email roles web-origins microprofile-jwt offline_access",
+    "credentials" : {
+    	"secret" : "${env.OIDC_CLIENT_SECRET}"
+    }
+}
+```
+
+Note that we have specified the `authentication-request-format` to be `request`, meaning, we are sending it by value. We
+have specified the signing algorithm to be *RS256*, and we are using the RSA key to sign the request object.We have also specified the *alg* and *enc* values to encrypt the request object. The request object JWT will be signed first and then encrypted
```
Comment (fjuma) [inline _posts/2024-07-03-oidc-cient-request.adoc]: Missing space before We

File: _posts/2024-07-03-oidc-cient-request.adoc [hunk 23]
```diff
@@ -0,0 +1,467 @@
...
+
+To obtain this example, clone the elytron-examples repository to your local machine:
+
+```
+git clone git@github.com:wildfly-security-incubator/elytron-examples.git
+```
+
+== Setting up the KeyPairs
+
+In order for us to sign the JWT's, we will need to use key pairs corresponding to the algorithms specified. These are tokens communicated between the client and the server and are used to sign and encrypt JWT's.
```
Comment (fjuma) [inline _posts/2024-07-03-oidc-cient-request.adoc]: "algorithms specified"

Is this needed?

Could we just reword this to something like we need to generate a key pair to be used to sign JWTs?

File: _posts/2024-07-03-oidc-cient-request.adoc [hunk 24]
```diff
@@ -0,0 +1,467 @@
...
+
+=== About Key Pairs
+
+Typically key pairs have 3 components:
```
Comment (fjuma) [inline _posts/2024-07-03-oidc-cient-request.adoc]: This might be confusing since pair is probably thought of as 2 things. Maybe mention the private key and public key first and then describe what the certificate is.

File: _posts/2024-07-03-oidc-cient-request.adoc [hunk 25]
```diff
@@ -0,0 +1,467 @@
...
+
+* *Public Key:* This is the part of the keypair that is shared with trusted parties. Typically, the OpenID provider shared a set of public keys with us which we use to encrypt our JWT's among other things. Public keys can be extracted from certificates.
+
+* *Certificate:* This is used to verify the signatures among other things. We share the certificates of keypairs we generate with the OpenID provider, and it contains the public key.
+
+Keypairs with public and private keys are known as *asymmetric* keys. Some keys are *symmetric*, meaning they only have 1 secret key that is used by both the client and the server. Each key has an algorithm that is used to generate it, such as, RSA, Elliptic Curve, etc. An example of a symmetric algorithm is HS256, where the same key is used to sign a JWT and verify the signature.
+
+=== Generating Key Pairs
+
+We can use the `keytool` tool in the cli to generate our keys and store them in a keystore file. We can generate keystore files of type `JKS` and `PKCS12` for our application. The general format for doing this is as follows:
```
Comment (fjuma) [inline _posts/2024-07-03-oidc-cient-request.adoc]: Maybe mention which type you'll use in this example.

File: _posts/2024-07-03-oidc-cient-request.adoc (hunk 19 above)
Comment (fjuma) [reply _posts/2024-07-03-oidc-cient-request.adoc]: If the latter, would be good to indicate that.

File: _posts/2024-07-03-oidc-cient-request.adoc [hunk 26]
```diff
@@ -0,0 +1,467 @@
...
+** For the *Valid Redirect URIs*, leave it empty for now. We will come back to edit it later.
+
+. Once you hit *Save*, you should see a new tab called *Keys* appear.
+
+. Navigate to the *Keys* tab for my client from the *Client details* page, 
+* click on the *Import* button located at the bottom of the screen. 
+** Under *Archive format*, choose *PKCS12* from the dropdown. 
+** Under *Key alias*, enter the alias of the RSA key you just created. 
+** Under *Store password* enter the password of the keystore you created.
+** under *Import file*, click on the *Browse* button to import the keystore file. Note that when we created the keystore, we specified a key
```
Comment (fjuma) [inline _posts/2024-07-03-oidc-cient-request.adoc]: Might be good to mention the name of the file that you are using for your example.

File: _posts/2024-07-03-oidc-cient-request.adoc [hunk 27]
```diff
@@ -0,0 +1,467 @@
...
+```
+Notice there’s a helm.yaml file in this directory with the following content:
+```
+build:
+  uri: https://github.com/wildfly-security-incubator/elytron-examples.git
+  contextDir: elytron-oidc-request
+deploy:
+  replicas: 1
+  volumes:
+    - name: saml-keystore-volume
```
Comment (fjuma) [inline _posts/2024-07-03-oidc-cient-request.adoc]: we should tweak the name since this was the name used for the saml example

File: _posts/2024-07-03-oidc-cient-request.adoc [hunk 28]
```diff
@@ -0,0 +1,467 @@
...
+      secret:
+        secretName: simple-webapp-saml-secret
```
Comment (fjuma) [inline _posts/2024-07-03-oidc-cient-request.adoc]: same here

File: _posts/2024-07-03-oidc-cient-request.adoc [hunk 29]
```diff
@@ -0,0 +1,467 @@
...
+an RSA key pair. For "ES" type keys, use Elliptical curve keys and as mentioned above, adjust the group name for the
+PKCS12 keystore keys to match the SHA value of the algorithms. "none" does not require a keystore and lastly, "HS" keys
+require a symmetric key, where the same secret hash is used by the client and the server to sign and verify respectively. 
+Not all algorithms are supported by all OpenID Providers. Review the documentation and/or the metadata for your OpenID provider to learn more about the supported algorithms. 
+
+== Summary
+
+This example has demonstrated how to secure a web application deployed to WildFly by sending the request parameters as a
+JWT. For more details on the `elytron-oidc-client` subsystem, please check out the
+https://docs.wildfly.org/33/Admin_Guide.html#Elytron_OIDC_Client[documentation] and for more details on OpenID Connect,
```
Comment (fjuma) [inline _posts/2024-07-03-oidc-cient-request.adoc]: Is there a link somewhere to the new section that you've added to the docs?

File: _posts/2024-07-03-oidc-cient-request.adoc (hunk 16 above)
Comment (PrarthonaPaul) [reply _posts/2024-07-03-oidc-cient-request.adoc]: time needed to build the pods once `helm install` is called. 
I have updated the description to reflect that.

File: _posts/2024-07-03-oidc-cient-request.adoc (hunk 17 above)
Comment (PrarthonaPaul) [reply _posts/2024-07-03-oidc-cient-request.adoc]: fixed!

File: _posts/2024-07-03-oidc-cient-request.adoc (hunk 18 above)
Comment (PrarthonaPaul) [reply _posts/2024-07-03-oidc-cient-request.adoc]: fixed!

File: _posts/2024-07-03-oidc-cient-request.adoc (hunk 19 above)
Comment (PrarthonaPaul) [reply _posts/2024-07-03-oidc-cient-request.adoc]: I have added a note that we don;t use the EC key, but if they want to use the eckey to sign, then the `ecKey` alias can be used instead when prompted.

File: _posts/2024-07-03-oidc-cient-request.adoc (hunk 20 above)
Comment (PrarthonaPaul) [reply _posts/2024-07-03-oidc-cient-request.adoc]: Just the one, but two of them are generated to show how it is done.

File: _posts/2024-07-03-oidc-cient-request.adoc (hunk 21 above)
Comment (PrarthonaPaul) [reply _posts/2024-07-03-oidc-cient-request.adoc]: fixed!

File: _posts/2024-07-03-oidc-cient-request.adoc (hunk 22 above)
Comment (PrarthonaPaul) [reply _posts/2024-07-03-oidc-cient-request.adoc]: fixed!

File: _posts/2024-07-03-oidc-cient-request.adoc (hunk 23 above)
Comment (PrarthonaPaul) [reply _posts/2024-07-03-oidc-cient-request.adoc]: fixed!

File: _posts/2024-07-03-oidc-cient-request.adoc (hunk 24 above)
Comment (PrarthonaPaul) [reply _posts/2024-07-03-oidc-cient-request.adoc]: combined certificate with public key

File: _posts/2024-07-03-oidc-cient-request.adoc (hunk 25 above)
Comment (PrarthonaPaul) [reply _posts/2024-07-03-oidc-cient-request.adoc]: fixed!

File: _posts/2024-07-03-oidc-cient-request.adoc (hunk 26 above)
Comment (PrarthonaPaul) [reply _posts/2024-07-03-oidc-cient-request.adoc]: added

File: _posts/2024-07-03-oidc-cient-request.adoc (hunk 27 above)
Comment (PrarthonaPaul) [reply _posts/2024-07-03-oidc-cient-request.adoc]: fixed!

File: _posts/2024-07-03-oidc-cient-request.adoc (hunk 28 above)
Comment (PrarthonaPaul) [reply _posts/2024-07-03-oidc-cient-request.adoc]: fixed!

File: _posts/2024-07-03-oidc-cient-request.adoc (hunk 29 above)
Comment (PrarthonaPaul) [reply _posts/2024-07-03-oidc-cient-request.adoc]: I have updated the link. 
Thanks @fjuma!

**Comment by fjuma** [comment]
@rsearls When you get a chance tomorrow, would you be able to do another pass through this to see if your comments have now been addressed by Prarthona's latest changes? Thanks!

**Comment by rsearls** [comment]
changes approved
Review complete.  No further changes recommended.

**Comment by fjuma** [comment]
Thanks @rsearls!

@PrarthonaPaul Please update the dates.

**Comment by PrarthonaPaul** [comment]
Thanks @rsearls and @fjuma 
I have updated the date!

**Review by fjuma — APPROVED** [review]
Thanks @PrarthonaPaul!
